docs/DeveloperGuide.md

```diff
@@ -4,34 +4,122 @@
+* [**`Data`**](#data-component): Holds the data of the app in memory.
+* [**`Command`**](#command-component): The command executor.
+
+### Core sequence
+Core sequence of code is written in [`FitTrack`](../src/main/java/fittrack/FitTrack.java) class.
+
+![Core structure](images/FitTrackOuter.svg "Outer Structure")
+
+![Inner structure](images/FitTrackCore.svg "Core Structure")
 
+### Creating a feedback for an invalid input
+Refer to [`CommandParser`](../src/main/java/fittrack/parser/CommandParser.java),
+[`InvalidCommand`](../src/main/java/fittrack/command/InvalidCommand.java),
+[`HelpCommand`](../src/main/java/fittrack/command/HelpCommand.java)
+classes.
+
+![Sequence of invalid command](images/InvalidCommand.svg "Sequence of invalid command")
+
+Given below is the Sequence Diagram for interactions within the Logic component for the execute("deletemeal 1") call.
+![Deletemeal command sequence](images/DeleteSequence.svg)
+
+### Storage Component
+Storage load and save functions are written in [`Storage`](../src/main/java/fittrack/storage/Storage.java) class.
+
+![Structure of Storage Load](images/StorageLoad.svg)
+The sequence diagram of the code for loading the file contents into each class.
+
+![Structure of Storage Save](images/StorageSave.svg)
+The sequence diagram of the code for saving data into file
+
+The `Storage` component,
+* can save user profile data in text format and load it back
+* can save meals in text format and load it back
+* can save workouts in text format and load it back
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+People who want to be healthy by managing their diet and workout.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+Fittrack is a health management application which allows users to record their diet and activity, and help them to reach the goal they have set.
+
+Fittrack also allows it's users to calculate their total calories spent in a day and receive suggestions
+on possible changes to their exercise, diet and lifestyle.
+
+Users will also be able to calculate key parameters of their profile like 
+BMI, ideal weight for their height and so on.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|Version| As a ... | I want to ...                                               | So that I can ...                                             |
+|--------|----------|-------------------------------------------------------------|---------------------------------------------------------------|
+|v1.0|new user| know how to use the product                                 | use the product                                               |
```

Comment:
Make the user comments more specific for the things that you want to implement. For example, a user knowing how to run your application doesn't really say much about what needs to be changed or updated